backend/src/main/scala/com/softwaremill/adopttapir/starter/files/FilesManager.scala

```diff
@@ -33,12 +34,11 @@ class FilesManager(config: StorageConfig):
       destination.toJava
     }
 
-  def deleteFilesAsStatedInConfig(destinationDir: IO[File]): IO[Unit] =
-    IO.blocking {
-      if config.deleteTempFolder then
-        for
-          dir <- destinationDir
-          _ <- dir.delete().pure[IO]
-        yield ()
-      else ().pure[IO]
-    }
+  def deleteFilesAsStatedInConfig(destinationDir: IO[File])(using cid: CorrelationId): IO[Unit] =
+    if config.deleteTempFolder then
+      for
+        dir <- destinationDir
+        deleted <- IO.blocking(dir.delete())
+        _ <- logger.info(s"The $dir deletion ${if (deleted) "succeeded" else "failed"}")
+      yield ()
+    else IO.unit
```

Comment:
**nitpick:** You can use `whenA` instead of `if`.

```scala
(
for
  dir <- destinationDir
  deleted <- IO.blocking(dir.delete())
  _ <- logger.info(s"The $dir deletion ${if (deleted) "succeeded" else "failed"}")
yield ()
).whenA(config.deleteTempFolder)
```

Reply:
Done. Thanks for the hint 💪